### ISSUE TITLE
Check bit string sizes statically

### ISSUE THREAD

**Issue by Janiczek** [gleam-lang/gleam#791]
Currently you can write
```
pub const x = <<100:size(1)>>
```
which to me seems nonsensical (it will just get truncated).

It gets compiled into Erlang's `<<100:1>>` which then gets interpreted as `<<0:1>>`.

**Perhaps it would make sense to raise a compiler error if the value bit length is larger than the advertised length after the colon?**

I'm not versed well enough in the Erlang ecosystem to say if there is some legitimate use for this truncation though. Might very well be, and then it would probably make sense to keep allowing it.

**Comment by itsgreggreg** [comment]
I don't think it truncates, I think it mods aka wraps around. That can be useful for that purpose:
```rust
pub fn bit_mod() -> Nil {
  [1, 2, 3, 4, 5, 6, 7, 8, 9]
  |> list.map(fn(i) { io.debug(<<i:2>>) })
  Nil
}

```
```
1> gleam_crash:bit_mod().
<<1:2>>
<<2:2>>
<<3:2>>
<<0:2>>
<<1:2>>
<<2:2>>
<<3:2>>
<<0:2>>
<<1:2>>
```

**Comment by NobbZ** [comment]
Truncating a number `n` by `b` bit is the same as `n % 2^b` where `%` is modulo and `^` "power of".
